### PR TITLE
CONSOLE-2371: [i18n] Externalize olm operand related strings

### PR DESCRIPTION
**Pull request by TheRealJon**
Externalize strings in `frontend/packages/operator-lifecycle-manager/src/components/operand`

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx [hunk 1]
```diff
@@ -834,7 +836,10 @@ export const DEPRECATED_CreateOperandForm: React.FC<OperandFormProps> = ({
       const k8sModel = modelFor(groupVersionKind);
       if (!k8sModel) {
         // eslint-disable-next-line no-console
-        console.warn('[Legacy CreateOperandForm] Cluster does not have resource', groupVersionKind);
+        console.warn(
+          t('olm~[Legacy CreateOperandForm] Cluster does not have resource'),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx:840]: No need to localize this string

File: frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx [hunk 2]
```diff
@@ -345,8 +347,8 @@ export const ProvidedAPIsPage = connect(inFlightStateToProps)((props: ProvidedAP
 
   const EmptyMsg = () => (
     <MsgBox
-      title="No Provided APIs Defined"
-      detail="This application was not properly installed or configured."
+      title={t('olm~No Provided APIs Defined')}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx:350]:
```suggestion
      title={t('olm~No provided APIs defined')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx [hunk 3]
```diff
@@ -396,12 +398,14 @@ export const ProvidedAPIsPage = connect(inFlightStateToProps)((props: ProvidedAP
...
-      filterLabel="Resources by name"
+      filterLabel={t('olm~Resources by name')}
       resources={firehoseResources}
       namespace={obj.metadata.namespace}
       canCreate={owned.length > 0}
       createProps={createProps}
-      createButtonText={owned.length > 1 ? 'Create New' : `Create ${owned[0].displayName}`}
+      createButtonText={t('olm~Create {{item}}', {
+        item: owned.length > 1 ? 'new' : owned[0].displayName,
+      })}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx:408]: I would make `Create new` and `Create {{item}}` separate messages

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts [hunk 4]
```diff
@@ -167,19 +168,25 @@ const getValidCapabilities = (descriptor: Descriptor<SpecCapability>, jsonSchema
   );
 
   if (invalid?.length) {
-    invalid.forEach((i) => {
+    invalid.forEach((invalidCapability) => {
       // eslint-disable-next-line no-console
       console.warn(
-        `[Operand Form] x-descriptor "${i}" is not compatible with schema property at ${descriptor.path} and will have no effect on the corresponding form field.`,
+        i18n.t(
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts:174]: No need to localize

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts [hunk 5]
```diff
@@ -167,19 +168,25 @@ const getValidCapabilities = (descriptor: Descriptor<SpecCapability>, jsonSchema
...
     });
   }
 
   if (deprecated?.length) {
-    deprecated.forEach((i) => {
+    deprecated.forEach((deprecatedCapability) => {
       // eslint-disable-next-line no-console
       console.warn(
-        `[Operand Form] x-descriptor "${i}" is deprecated and will have no effect on the form field corresponding to ${descriptor.path}`,
+        i18n.t(
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts:186]: No need to localize

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts [hunk 6]
```diff
@@ -199,7 +206,9 @@ export const descriptorsToUISchema = (
       if (!schemaForDescriptor) {
         // eslint-disable-next-line no-console
         console.warn(
-          '[OperandForm] SpecDescriptor path references a non-existent schema property:',
+          i18n.t(
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts:209]: No need to localize

File: frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx [hunk 7]
```diff
@@ -1217,16 +1219,16 @@ export const DEPRECATED_CreateOperandForm: React.FC<OperandFormProps> = ({
...
               >
-                {error || 'Fix above errors'}
+                {error || t('olm~Fix above errors')}
               </Alert>
             )}
             <div style={{ paddingBottom: '30px' }}>
               <ActionGroup className="pf-c-form">
                 <Button onClick={submit} type="submit" variant="primary">
-                  Create
+                  {t('olm~Create')}
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx:1228]:
```suggestion
                  {t('public~Create')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx [hunk 8]
```diff
@@ -1217,16 +1219,16 @@ export const DEPRECATED_CreateOperandForm: React.FC<OperandFormProps> = ({
...
                 </Button>
                 <Button onClick={history.goBack} variant="secondary">
-                  Cancel
+                  {t('olm~Cancel')}
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/DEPRECATED_operand-form.tsx:1231]:
```suggestion
                  {t('public~Cancel')}
```

**Review by rhamilto — COMMENTED** [review]
`Create` and `Cancel` are already in `public` namespace, so no reason to add them to `olm`, too.

**Comment by TheRealJon** [comment]
/hold

For qe and docs approval

**Comment by rhamilto** [comment]
@TheRealJon rebase required.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/hold cancel

@TheRealJon FYI, needs rebase

**Comment by TheRealJon** [comment]
@spadgett rebased. Needs lgtm again.

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**